### PR TITLE
ThreeCoin router fix + AAVE and LINK routing improvements.ts

### PR DESCRIPTION
**Pull request by Dusknight4**
Added additional routing bases for ThreeCoin

---

## PR-Codex overview

> The following files were skipped due to too many changes: `config/router/src/common-bases.ts`, `config/router/src/bases-to-check-trades-against.ts`, `config/router/src/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by matthewlilley** [comment]
@Dusknight4 would you be able to do me a favour and list the token addresses for TWO, THREE & FOUR? I'll review and make updates to the PR so we can test once you've done that.

**Comment by Dusknight4** [comment]
@matthewlilley here's a link to my scope statement: https://docs.google.com/document/d/1gdeuzhLRZ_xMDvXIcJ0nbrkY625i6xMBXHPS5QCBK2Q/edit?usp=sharing

TwoCoin Eth address: 0x1e0275806C3CD0bDb5C99916A064d36b5e8eAE8d
Threecoin Matic Address: 0x9B034262e0095210ab9ddec60199741a8a1FbFe7
Fourcoin FTM address: 0x9E8862e39496BD336565Dd15cE2C0B90bc7dc121

File: config/router/src/common-bases.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  SUSHI,
+  Token,
+  USDC,
+  USDT,
+  WBTC,
+  WETH9,
+  WNATIVE,
+} from '@sushiswap/currency'
+
+export const COMMON_BASES = {
```
Comment (Dusknight4) [inline config/router/src/common-bases.ts:19]: Why not expand Common Bases to include more tokens on networks with fewer fees? Namely Sushi and XSushi. This should increase volume and trading fees that the tokens earn.

File: config/router/src/additional-bases.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    [KLIMA_ADDRESS[ChainId.POLYGON]]: [BCT[ChainId.POLYGON]],
+    // THREE
+    ['0x9B034262e0095210ab9ddec60199741a8a1FbFe7']: [
+      AAVE[ChainId.POLYGON],
+      LINK[ChainId.POLYGON],
+      MANA[ChainId.POLYGON],
+      SNX[ChainId.POLYGON],
+      CRV[ChainId.POLYGON],
+      YFI[ChainId.POLYGON],
+      // ENJ[ChainId.POLYGON], // could not find on polygon
```
Comment (Dusknight4) [inline config/router/src/additional-bases.ts:91]: Above needs routing for XSushi and Sushi unless they're being added on Polygon as common bases.

File: config/router/src/additional-bases.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  [ChainId.FANTOM]: {
+    [FRAX_ADDRESS[ChainId.FANTOM]]: [FXS[ChainId.FANTOM]],
+    [FXS_ADDRESS[ChainId.FANTOM]]: [FRAX[ChainId.FANTOM]],
+    // FOUR
+    ['0x9E8862e39496BD336565Dd15cE2C0B90bc7dc121']: [
+      LINK[ChainId.FANTOM],
+      SNX[ChainId.FANTOM],
+      CRV[ChainId.FANTOM],
+      YFI[ChainId.FANTOM],
+    ],
```
Comment (Dusknight4) [inline config/router/src/additional-bases.ts:107]: Above needs routing for Sushi and XSushi unless they're being added to Fantom as common bases.

File: config/router/src/additional-bases.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      AAVE[ChainId.ETHEREUM],
+      MKR[ChainId.ETHEREUM],
+      SNX[ChainId.ETHEREUM],
+      CRV[ChainId.ETHEREUM],
+      YFI[ChainId.ETHEREUM],
+      ENJ[ChainId.ETHEREUM],
+      COMP[ChainId.ETHEREUM],
+      GALA[ChainId.ETHEREUM],
+      XSUSHI[ChainId.ETHEREUM],
+    ],
```
Comment (Dusknight4) [inline config/router/src/additional-bases.ts:76]: Above needs LINK added as well.

**Pull request merged**